File: solutions/swift/sublist/2/Sources/Sublist/Sublist.swift
enum Relation { case equal, sublist, superlist, unequal }

func classifier(listOne: [Int], listTwo: [Int]) -> Relation {
  listOne == listTwo
    ? .equal
    : listOne.contains(listTwo)
      ? .superlist
      : listTwo.contains(listOne)
        ? .sublist : .unequal
}

File: solutions/clojure/sublist/1/src/sublist.clj
(ns sublist)

(defn classify [coll1 coll2]
  (letfn [(contains-sublist? [coll candidate] (not= -1 (java.util.Collections/indexOfSubList coll candidate)))]
    (cond
      (= coll1 coll2) :equal
      (contains-sublist? coll1 coll2) :superlist
      (contains-sublist? coll2 coll1) :sublist
      :else :unequal)))
